src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,69 @@
+package duke;
+
+/**
+ * Handles making sense of user input.
+ */
+
+public class Parser {
+
+    public Parser() {
+    }
+
+    /**
+     * Converts user input into a command of the appropriate type depending on the leading word
+     * of the user input.
+     * @param input User input.
+     * @return duke.Command object corresponding to input.
+     * @throws DukeException When user input cannot be parsed.
+     */
+    public Command parse(String input) throws DukeException {
```

Comment:
I like the use of `Constants.CommandType` enum together with switch-case, make it easy to understand the different commands.